src/EFCore.PG/Query/ExpressionTranslators/Internal/NpgsqlStringMethodTranslator.cs

```diff
@@ -290,5 +288,10 @@ string EscapeLikePattern(string pattern)
 
             return builder.ToString();
         }
+
+        SqlExpression GenerateOneBasedIndexExpression([NotNull] SqlExpression expression)
+            => expression is SqlConstantExpression constant
+                ? _sqlExpressionFactory.Constant(Convert.ToInt32(constant.Value) + 1, constant.TypeMapping)
```

Comment:
Maybe just cast to `int`, no? The `Substring` method has an `int` as its first parameter. Therefore, nothing other could be stored by `Value`.

Reply:
Yeah, in this case you're right, but this is a somewhat general method that can in theory work for other cases where there are other options... So safer to leave it like this in case it gets used in some other context (and it doesn't really matter :)).